src/coreclr/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.CoreCLR.cs

```diff
@@ -147,7 +147,7 @@ private static IntPtr ResolveUnmanagedDllUsingEvent(string unmanagedDllName, Ass
 
             AssemblyLoadContext? loadContextForAssembly = null;
 
-            RuntimeAssembly? rtAsm = assembly as RuntimeAssembly;
+            RuntimeAssembly? rtAsm = GetRuntimeAssembly(assembly);
```

Comment:
Ok, I'll put this fix in a separate PR with the appropriate tests.

Reply:
The same bug exists in `ValidateAssemblyNameWithSimpleName`. I filed a new issue for it: https://github.com/dotnet/runtime/issues/53181

But it might be a good idea to fix both at the same time.

Reply:
No problem) I'll do it